### PR TITLE
Use Popen.wait to wait for mouse grid subprocess to exit

### PR DESCRIPTION
**Pull request by dmakarov**
Mouse grid improvements

## Description

These changes affect how Caster manages the processes that it starts
to display mouse grid windows.  The current algorithm is to query the
system if there exist any application with certain names, depending on
the grid used.  The grid subprocesses are objects of subprocess.Popen
class, hence it's more natural to use Popen methods to manage the
subprocesses.

Additional changes migrate mouse commands to OS agnostic methods of
pyinput package.

## Related Issue

N/A

## Motivation and Context

These changes are believed to improve the code by streamlining the grid subprocess management by using the functionality available via Python standard library.

## How Has This Been Tested

I tested by running Caster on macOS and using Caster's mouse grids functionality.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue or bug)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Renamed existing command phrases (we discourage this without a strong rationale).

## Checklist

- [x] I have read the CONTRIBUTING document.
- [x] My code follows the code style of this project.
- [x] I have checked that my code does not duplicate functionality elsewhere in Caster.
- [x] My code implements all the features I wish to merge in this pull request.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests pass.

## Maintainer/Reviewer Checklist

- [ ] Basic functionality has been tested and works as claimed.
- [ ] New documentation is clear and complete.
- [ ] Code is clear and readable.

### CODE REVIEW

**Comment by LexiconCode** [comment]
Looking forward to testing this on Windows!

Dragonfly utilizes pyinput and has actions to control the Mouse. Are there reasons that Dragonfly's mouse actions can't be used?

**Comment by dmakarov** [comment]
Not really, will migrate to Dragonfly.

File: castervoice/lib/merge/state/stackitems.py [hunk 1]
```diff
@@ -63,10 +63,13 @@ def put_time_action(self):
         if settings.SETTINGS["miscellaneous"]["print_rdescripts"] and self.show:
             # formats rdescript with the given data
             try:
-                rd = self.rdescript % self.dragonfly_data
-            except:
+                rd = self.rdescript % self.dragonfly_data if self.dragonfly_data else self.rdescript
```
Comment (LexiconCode) [inline castervoice/lib/merge/state/stackitems.py]: The bug that this addresses I could not reproduce with the original settings window but I did have it occur with your QT PR. Hopefully this resolves an issue

**Comment by dmakarov** [comment]
removed dependency on pyinput.

**Comment by LexiconCode** [comment]
Thank you for your time with this. It brings the Caster project closer to cross-platform support!

**Pull request merged**